cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/connector/ConnectorSink.java

```diff
@@ -66,18 +66,23 @@ public void prepareRun(BatchSinkContext context) throws Exception {
   }
 
   @Override
-  public void transform(StructuredRecord input, Emitter<KeyValue<NullWritable, Text>> emitter) throws Exception {
+  public void transform(KeyValue<String, StructuredRecord> input, Emitter<KeyValue<NullWritable, Text>> emitter)
+    throws Exception {
+    StructuredRecord taggedInput = input.getValue();
     if (writeSchema) {
```

Comment:
Do you mean `writeSchema` currently is always true? I have removed it as we support multiple schemas now and we will need schema to be written everytime.